asset/mock.go

```diff
@@ -49,3 +59,39 @@ func RandID(t testing.TB) ID {
 
 	return a
 }
+
+// RandAsset creates a random asset of the given type for testing.
+func RandAsset(t testing.TB, assetType Type) *Asset {
```

Comment:
I was thinking that these could be moved into `internal/test`, but I'm pretty sure that would cause a dependency cycle.

Reply:
Yes, that was the initial goal but that would indeed introduce a dependency cycle unless we declared all tests in `asset` in a different test package (e.g. `package asset_test` instead of `package asset`). But then we couldn't test non-exported methods anymore. So I would say this is an acceptable compromise.